### DISCUSSION TITLE
How to fix invalid frame errors?

### DISCUSSION THREAD

**Discussion by devintly** [OvenMediaLabs/OvenMediaEngine#1365] [Help]
Hello, can you please help me with these errors:
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/72499082/72002bcd-ad31-4816-af93-ac277bd302df)

I'm streaming with webrtc whip protocol with latest obs v30.0.0 beta2

My server.xml outputProfile:
`<Name>bypass_stream</Name>
							<OutputStreamName>${OriginStreamName}</OutputStreamName>
							<Playlist>
								<Name>for Webrtc</Name>
								<FileName>abr</FileName>
								<Options>
									<WebRtcAutoAbr>false</WebRtcAutoAbr> 
								</Options>
								<Rendition>
									<Name>1080p</Name>
									<Video>bypass_video</Video>
									<Audio>bypass_audio</Audio>
								</Rendition>
								<Rendition>
									<Name>720p</Name>
									<Video>video_720</Video>
									<Audio>bypass_audio</Audio>
								</Rendition>
							</Playlist>
							<Encodes>
								<Audio>
									<Name>bypass_audio</Name>
									<Bypass>true</Bypass>
								</Audio>
								<Video>
									<Name>bypass_video</Name>
									<Bypass>true</Bypass>
								</Video>
								<Video>
									<Name>video_720</Name>
									<Codec>h264</Codec>
									<Bitrate>4096000</Bitrate>
									<Framerate>30</Framerate>
									<Width>1280</Width>
									<Height>720</Height>
									<Preset>medium</Preset>
								</Video>
							</Encodes>`
My OBS settings:
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/72499082/6c2338b2-96cb-4be9-a839-762bafffe67c)
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/72499082/8511da2a-e748-4525-80e9-ee2e5f3d20f7)

**Comment by basisbit** [comment]
in the additional options of the ffmpeg encoder in OBS, enter `bf=0 bframes=0`

**Comment by devintly** [comment]
Tried this, but problem still exist(
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/72499082/3af0f9af-78b8-4565-b552-d44886f36efc)
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/72499082/4ccff3db-9bb3-40f2-b70d-efd1bbf458c2)

**Comment by getroot** [comment]
It is very likely that this is caused by packet loss. Since OBS does not support TURN, it will only communicate using UDP. To check if this assumption is correct, it would be a good idea to test with very low bitrates (500kb).

And please share information about the OBS you use. (download link)

**Reply by devintly** [reply]
Yes, you're right. With 500kb are no errors. So what can we do with this?
My OBS: https://github.com/obsproject/obs-studio/releases/tag/30.0.0-beta2

**Reply by getroot** [reply]
Oh, WHIP is included in OBS Beta. That's good news. It appears that udp packet loss is occurring in your network section (OBS -> OME). This will be naturally resolved when OBS supports TURN, allowing communication via TCP. When I looked in the past, OBS did not support TURN yet. I will check this out soon.

**Reply by naanlizard** [reply]
For obs, I recommend trying VBR instead of CBR, and some higher cpu preset. Then lower your max bitrate. You’ll need to play with each of those settings to find a good mix but you should be able to use far less while looking good

**Reply by basisbit** [reply]
@naanlizard VBR tends to be less reliable when congestion happens at the last mile of the ISP's network or when the congestion happens in the home network of the streamer. The more steady behavior of CBR tends to keep other TCP streams from growing by its congestion control algorithms. When the packet loss is mainly happening at a congested peering/transit, then VBR is similarly affected and low bitrate UDP streams tend to be similarly affected, so this seems to not be the case here.

TLDR: stick with CBR for live video streaming, lower your bitrate until it is reliably supported by your internet connection and prefer wired connection over wireless connection. Also, try to pick an OvenMediaEngine server which is located in your region of the world, not one that is far away from you.

**Reply by naanlizard** [reply]
I've heard people say such, but in my experience VBR generally works better for streamers, especially when they're not great at tuning their bitrate against quality. Either way 10mbps is way high for most content and not using ultrafast will help quite a bit
